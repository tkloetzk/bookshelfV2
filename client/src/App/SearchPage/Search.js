import Grid from '@material-ui/core/Grid'
import React from 'react'
import { makeStyles } from '@material-ui/styles'
import { useSelector } from 'react-redux'
import Results from '../Results/Results'
import SearchBar from './SearchBar/SearchBar'

const useStyles = makeStyles(theme => ({
  container: {
    padding: theme.spacing(2),
    margin: 0,
    width: '100%',
  },
}))

export default function SearchPage() {
  const classes = useStyles()
  const [booklist, setBooklist] = React.useState([])
  const bookshelf = useSelector(state => state.bookshelf.bookshelf)

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      spacing={4}
      className={classes.container}
      data-testid="searchView"
    >
      <SearchBar setBooklist={setBooklist} bookshelf={bookshelf} />
      <Grid item xs={12}>
        <Results booklist={booklist} />
      </Grid>
    </Grid>
  )
}
