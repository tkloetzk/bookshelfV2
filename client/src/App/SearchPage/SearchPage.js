import Grid from '@material-ui/core/Grid'
import React from 'react'
import { makeStyles } from '@material-ui/styles'
import { useSelector } from 'react-redux'
import Results from '../Results/Results'
import SearchBar from './SearchBar/SearchBar'
import remove from 'lodash/remove'

const useStyles = makeStyles(theme => ({
  container: {
    padding: theme.spacing(2),
    margin: 0,
    width: '100%',
  },
}))

export default function SearchPage() {
  const classes = useStyles()
  const [booklist, setBooklist] = React.useState([])
  const bookshelf = useSelector(state => state.bookshelf.bookshelf)

  function handleSave(book) {
    remove(booklist, {
      isbn: book.isbn,
    })
    setBooklist([...booklist, book])
  }

  function handleDelete(book) {
    remove(booklist, {
      isbn: book.isbn,
    })
    setBooklist([...booklist])
  }

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      spacing={4}
      className={classes.container}
      data-testid="searchView"
    >
      <SearchBar
        setBooklist={setBooklist}
        bookshelf={bookshelf}
        booklist={booklist}
      />
      <Grid item xs={12}>
        <Results
          booklist={booklist}
          handleSave={handleSave}
          handleDelete={handleDelete}
        />
      </Grid>
    </Grid>
  )
}
