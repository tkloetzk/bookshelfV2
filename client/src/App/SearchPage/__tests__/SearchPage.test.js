import { MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { fireEvent, render, wait } from '@testing-library/react'
import muiTheme from '../../../config/themeConfig'
import SearchPage from '../SearchPage'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'

const mockStore = configureMockStore()

describe('SearchPage', () => {
  let store
  beforeEach(() => {
    store = mockStore({
      bookshelf: { bookshelf: [] },
    })
  })
  describe('render', () => {
    it('should render as expected', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchPage />
          </MuiThemeProvider>
        </Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
