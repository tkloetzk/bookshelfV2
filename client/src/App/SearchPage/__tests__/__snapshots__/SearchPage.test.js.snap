// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage render should render as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    data-testid="searchView"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            data-testid="searchBar"
            rows="1"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-143 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-144"
              style="width: 0.01px;"
            >
              <span>
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
        data-testid="searchButton"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search
        </span>
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      />
    </div>
  </div>
</DocumentFragment>
`
